src/AWS.Deploy.Recipes/CdkTemplates/AspNetAppElasticBeanstalkLinux/Generated/UpdateBeanstalkEnvironment/OnEventLambda/OnEvent.py

```diff
@@ -0,0 +1,27 @@
+import boto3
+
+client = boto3.client('elasticbeanstalk')
+
+
+def handler(event, context):
+    request_type = event['RequestType'].lower()
+    if request_type == 'create' or request_type == 'update':
+        return on_create_or_update(event)
+    if request_type == 'delete':
+        return 
+    raise Exception(f'Invalid request type: {request_type}')
+
+
+def on_create_or_update(event):
+    props = event["ResourceProperties"]
+    update_environment(props['ApplicationName'], props['EnvironmentName'], props['VersionLabel'])
+    physical_id = get_physical_id(props)
+    return {'PhysicalResourceId': physical_id}
+
+
+def update_environment(applicationName, environmentName, versionLabel):
+    response = client.update_environment(ApplicationName = applicationName, EnvironmentName = environmentName,VersionLabel=versionLabel)
+
+
+def get_physical_id(props):
+    return props['ApplicationName'] + '-' + props['EnvironmentName'] + '-' + props['VersionLabel']
```

Comment:
Based on the post you sent me the physical id should not change over redeployments.  I think you just need to remove `VersionLabel` and I would add a constant prefix some to basically namespace it.

```
return `EBUpdate-` + props['ApplicationName'] + '-' + props['EnvironmentName']
```


https://medium.com/cyberark-engineering/custom-resources-with-aws-cdk-d9a8fad6b673#0b26

```
IMPORTANT: For the physical resource id, you need to use something that will not change over deployments. I simply used the object full name which is unique and unchanging. If your physical id changes your resource will be deleted on an update, since CloudFormation considers these different resources.
```

Reply:
Fixed